Submitted By:            Bruce Dubbs <bdubbs_at_linuxfromscratch_dot_org>
Date:                    2015-10-19
Initial Package Version: 1.0.1
Upstream Status:         None.  BLFS specific.
Origin:                  BLFS
Description:             Updates Makefile enabling parameter passing.
                         Rediffed for version 1.0.1
Updates:                 Rediffed for version 1.0.3 (P. Labastie)

diff -Naur gptfdisk-1.0.3.old/Makefile gptfdisk-1.0.3.new/Makefile
--- gptfdisk-1.0.3.old/Makefile	2017-07-28 03:41:20.000000000 +0200
+++ gptfdisk-1.0.3.new/Makefile	2017-07-31 14:33:13.656371245 +0200
@@ -1,29 +1,39 @@
-CC=gcc
-CXX=g++
-CFLAGS+=-D_FILE_OFFSET_BITS=64
-#CXXFLAGS+=-Wall -D_FILE_OFFSET_BITS=64 -D USE_UTF16
-CXXFLAGS+=-Wall -D_FILE_OFFSET_BITS=64
-LDFLAGS+=
-LIB_NAMES=crc32 support guid gptpart mbrpart basicmbr mbr gpt bsd parttypes attributes diskio diskio-unix
-MBR_LIBS=support diskio diskio-unix basicmbr mbrpart
-LIB_OBJS=$(LIB_NAMES:=.o)
-MBR_LIB_OBJS=$(MBR_LIBS:=.o)
-LIB_HEADERS=$(LIB_NAMES:=.h)
-DEPEND= makedepend $(CXXFLAGS)
+ifdef ICU
+ USE_ICU     = -DUSE_UTF16
+ EXTRA_LIBS  = -licuio -licuuc
+endif
+
+ifdef POPT
+ EXTRA_LIBS += -lpopt
+ SGDISK      = sgdisk
+endif
+
+CC           = gcc
+CXX          = g++
+CFLAGS      += -D_FILE_OFFSET_BITS=64
+CXXFLAGS    += -Wall -D_FILE_OFFSET_BITS=64 $(USE_ICU)
+
+LDFLAGS     +=
+LIB_NAMES    = crc32 support guid gptpart mbrpart basicmbr mbr \
+               gpt bsd parttypes attributes diskio diskio-unix
+LIB_OBJS     = $(LIB_NAMES:=.o)
+LIB_HEADERS  = $(LIB_NAMES:=.h)
 
-all:	cgdisk gdisk sgdisk fixparts
+MBR_LIBS     = support diskio diskio-unix basicmbr mbrpart
+MBR_LIB_OBJS = $(MBR_LIBS:=.o)
+
+DEPEND       = makedepend $(CXXFLAGS)
+
+all: cgdisk gdisk fixparts $(SGDISK)
 
 gdisk:	$(LIB_OBJS) gdisk.o gpttext.o
-	$(CXX) $(LIB_OBJS) gdisk.o gpttext.o $(LDFLAGS) -luuid -o gdisk
-#	$(CXX) $(LIB_OBJS) gdisk.o gpttext.o $(LDFLAGS) -licuio -licuuc -luuid -o gdisk
+	$(CXX) $(LIB_OBJS) gdisk.o gpttext.o $(LDFLAGS) $(EXTRA_LIBS) -luuid -o gdisk
 
 cgdisk: $(LIB_OBJS) cgdisk.o gptcurses.o
-	$(CXX) $(LIB_OBJS) cgdisk.o gptcurses.o $(LDFLAGS) -luuid -lncursesw -o cgdisk
-#	$(CXX) $(LIB_OBJS) cgdisk.o gptcurses.o $(LDFLAGS) -licuio -licuuc -luuid -lncurses -o cgdisk
+	$(CXX) $(LIB_OBJS) cgdisk.o gptcurses.o $(LDFLAGS) $(EXTRA_LIBS) -luuid -lncursesw -o cgdisk
 
 sgdisk: $(LIB_OBJS) sgdisk.o gptcl.o
-	$(CXX) $(LIB_OBJS) sgdisk.o gptcl.o $(LDFLAGS) -luuid -lpopt -o sgdisk
-#	$(CXX) $(LIB_OBJS) sgdisk.o gptcl.o $(LDFLAGS) -licuio -licuuc -luuid -lpopt -o sgdisk
+	$(CXX) $(LIB_OBJS) sgdisk.o gptcl.o $(LDFLAGS) $(EXTRA_LIBS) -luuid -lpopt -o sgdisk
 
 fixparts: $(MBR_LIB_OBJS) fixparts.o
 	$(CXX) $(MBR_LIB_OBJS) fixparts.o $(LDFLAGS) -o fixparts
@@ -37,6 +47,11 @@
 clean:	#no pre-reqs
 	rm -f core *.o *~ gdisk sgdisk cgdisk fixparts
 
+install: all 
+	mkdir -p $(DESTDIR)/sbin $(DESTDIR)/usr/share/man/man8
+	cp gdisk cgdisk fixparts $(SGDISK) $(DESTDIR)/sbin
+	cp *.8 $(DESTDIR)/usr/share/man/man8
+
 # what are the source dependencies
 depend: $(SRCS)
 	$(DEPEND) $(SRCS)
